src/aliceVision/keyframe/README.md

```diff
@@ -0,0 +1,114 @@
+# Keyframe Selection
+
+This module provides several methods to perform a keyframe selection.
+
+The goal of the keyframe selection is to extract, from an input video or an input sequence of images, keyframes.
+Two methods are currently supported:
+- a **regular** selection method, which selects keyframes regularly across the input video / sequence according to a set of parameters;
+- a **smart** selection method, which analyses the sharpness and motion of all the frames to select those which are deemed the most relevant (a frame is considered relevant if it contains significant motion in comparison to the last selected keyframe while being as sharp as possible).
+
+The selected keyframes can be written as JPG, PNG or EXR images, and the storage data type can be specified when the EXR file extension is selected.
+
+The keyframe selection module supports the following inputs:
+- a path to a video file (e.g. "/path/to/video.mp4")
+- a path to a folder containing images (e.g. "/path/to/folder/")
+- a path to a folder containing images with a regular expression (e.g. "/path/to/folder/*.exr")
+
+Camera rigs are also supported.
+
+## Regular selection method
+
+The regular selection samples frames regularly over time with respect to some user-provided constraints:
+- `minFrameStep`: the minimum number of frames between two selected keyframes;
+- `maxFrameStep`: the maximum number of frames between two selected keyframes;
```

Comment:
not clear to me what the role of the two parameters is: if it is a regular sampling of the frames (e.g. 1 frame every 5) why do we need 2 parameters? Or there is still a sharpness score influencing the choice of the frame? Like take  the sharpest frame that is at least `minFrameStep` far from the previous and at most `maxFrameStep` far from the previous one. Is it like this?

Reply:
There is no influence of the sharpness or the optical flow in the regular method, it really is just a regular sampling that can be modulated. 
`maxFrameStep` is mostly useful when you want to combine a relatively strict sampling with a maximum number of output frames. It is basically here to take precedence over `maxNbOutFrames`.

The expected behaviour for the regular selection depending on the set parameters can be summed up as follows:

- If only `minFrameStep` is set, the whole sequence will be sampled and a keyframe will be selected every `minFrameStep`. E.g: your sequence has 2000 frames, `minFrameStep = 100`, you will get 21 keyframes: with exactly 100 frames between them.

- If `minFrameStep` and `maxNbOutFrames` are set, you will never have less than `minFrameStep` between your keyframes, but you might have more for the whole sequence to be sampled while respecting `maxNbOutFrames`. E.g: your sequence has 2000 frames, `minFrameStep = 100` and `maxNbOutFrames = 10`: you will get 10 keyframes with 222 frames between them, so both `maxNbOutFrames` and `minFrameStep` are respected. If your sequence has 500 frames and `minFrameStep = 100` / `maxNbOutFrames = 10`, you will get 6 keyframes with 100 frames between them. No matter the value of the parameters, the entire sequence will be sampled.

- If `minFrameStep` and `maxFrameStep` are both set but `maxNbOutFrames` isn't, then the step between two keyframes will be exactly between `minFrameStep` and `maxFrameStep`. If `minFrameStep = 100` and `maxFrameStep = 200` without other constraints, it is equivalent to setting `minFrameStep = 150` (so, not the most useful use-case).

- If `minFrameStep`, `maxFrameStep` and `maxNbOutFrames` are all set, then `maxFrameStep` prevents the step between two frames from increasing too much to respect `maxNbOutFrames`. If we take again the example of the sequence with 2000 frames, if we have `minFrameStep = 100`, `maxFrameStep = 150` and `maxNbOutFrames = 10`, then we will get 10 keyframes with 150 frames between each, and the sampling will stop before reaching the end of the sequence so that all the constraints are respected. In the same example, if `maxFrameStep = 300`, then we will end up with 10 keyframes with 222 frames between them, and the whole sequence will be sampled.
`maxFrameStep` **ensures** that there will not be way more frames between two keyframes that we might want. 

I hope this is clear enough (and if you have any suggestion to make that clearer in the README, please let me know)!

Reply:
Thanks for the explanation! :+1: 
Ok, it's not that trivial either then :-)

I think most of the users would be interested in the two basic settings:
1. taking one frame every `minFrameStep` all along the video
2. taking `maxNbOutFrames` equally spaced along the video.

Then all the others are more fine-tuning settings for advanced needs.
So at least in the documentation, we can highlight those two primary cases and then make a sub-section (`Advanced sampling`) where we explain all the other combinations and possibilities to tune the sampling further.
And since you already took the time to kindly write it you can re-use what you write with the examples because it is much more clear.

What do you think?

Reply:
Sounds good to me! I updated the `README.md` accordingly with these clarifications and examples.
Thanks for the feedback!